test/unittest/unittest_base_properties.cpp

```diff
@@ -106,6 +106,38 @@ class PtrOfBanana : public nntrainer::Property<int *> {
   static constexpr const char *key = "ptr_banana";
   using prop_tag = nntrainer::ptr_prop_tag;
 };
+
+/**
+ * @brief Enuminformation of BananaType;
+ *
+ */
+struct BananaEnumInfo {
+  /**
+   * @brief underlying enum
+   *
+   */
+  enum class Enum {
+    Cavendish = 0,
+    Plantain = 1,
+    Manzano = 2,
+  };
+
+  static constexpr std::initializer_list<Enum> EnumList = {
+    Enum::Cavendish, Enum::Plantain, Enum::Manzano};
+
+  static constexpr const char *EnumStr[] = {"Cavendish", "Plantain", "Manzano"};
```

Comment:
I wonder we need to set this every enum variable then.

Reply:
Yes, there is no way to get string information except dirty methods (using some hacky macros)

Reply:
this is ok for now but consider moving to map/unoredered_map for enums where values are not continuously defined (which happens in layer types).

Reply:
Right, to iterate over it,
below information is forced to be inserted as well.
```
static constexpr std::initializer_list<Enum> EnumList = {
    Enum::Cavendish, Enum::Plantain, Enum::Manzano};
```

I used two primitives to make it constexpr but maps also works :)